File: fibo.py
from collections import OrderedDict

class LRUCache:
    def __init__(self, capacity):
        self.capacity = capacity
        self.cache = OrderedDict()

    def get(self, key):
        if key in self.cache:
            self.cache.move_to_end(key)
            return self.cache[key]
        return -1

    def put(self, key, value):
        if key in self.cache:
            self.cache.move_to_end(key)
        self.cache[key] = value
        if len(self.cache) > self.capacity:
            self.cache.popitem(last=False)

class Employee:
    def __init__(self, name, age, salary):
        self.name = name
        self.age = age
        self.salary = salary

    def __str__(self):
        return f"{self.name} ({self.age}, {self.salary})"

if __name__ == "__main__":
    employees = [
        Employee("Alice", 30, 50000),
        Employee("Bob", 25, 60000),
        Employee("Charlie", 35, 50000),
        Employee("David", 28, 60000),
        Employee("Eve", 32, 55000)
    ]

    # Sort by salary descending, then age ascending, then name ascending
    employees.sort(key=lambda e: (-e.salary, e.age, e.name))

    for e in employees:
        print(e)

File: fibo.java
import java.util.*;

class LRUCache {
    private LinkedHashMap<Integer, Integer> map;
    private int capacity;

    public LRUCache(int capacity) {
        this.capacity = capacity;
        this.map = new LinkedHashMap<Integer, Integer>(capacity, 0.75f, true) {
            protected boolean removeEldestEntry(Map.Entry<Integer, Integer> eldest) {
                return size() > capacity;
            }
        };
    }

    public int get(int key) {
        return map.getOrDefault(key, -1);
    }

    public void put(int key, int value) {
        map.put(key, value);
    }
}

class Employee {
    String name;
    int age;
    double salary;

    Employee(String name, int age, double salary) {
        this.name = name;
        this.age = age;
        this.salary = salary;
    }

    @Override
    public String toString() {
        return name + " (" + age + ", " + salary + ")";
    }
}

class EmployeeComparator implements Comparator<Employee> {
    @Override
    public int compare(Employee e1, Employee e2) {
        // 1. Salary descending
        int salaryCompare = Double.compare(e2.salary, e1.salary);
        if (salaryCompare != 0) {
            return salaryCompare;
        }
        // 2. Age ascending
        int ageCompare = Integer.compare(e1.age, e2.age);
        if (ageCompare != 0) {
            return ageCompare;
        }
        // 3. Name alphabetical (ascending)
        return e1.name.compareTo(e2.name);
    }
}

class Main {
    public static void main(String[] args) {
        List<Employee> employees = Arrays.asList(
            new Employee("Alice", 30, 50000),
            new Employee("Bob", 25, 60000),
            new Employee("Charlie", 35, 50000),
            new Employee("David", 28, 60000),
            new Employee("Eve", 32, 55000)
        );

        Collections.sort(employees, new EmployeeComparator());

        for (Employee e : employees) {
            System.out.println(e);
        }
    }
}
